src/ninja.cc

```diff
@@ -1541,18 +1546,78 @@ bool NinjaMain::EnsureBuildDirExists() {
   return true;
 }
 
+std::unique_ptr<Jobserver::Client> NinjaMain::SetupJobserverClient(
+    Status* status) {
+  // Empty result by default.
+  std::unique_ptr<Jobserver::Client> result;
+
+  // If dry-run or explicit job count, don't even look at MAKEFLAGS
+  if (config_.disable_jobserver_client)
+    return result;
+
+  const char* makeflags = getenv("MAKEFLAGS");
+  if (!makeflags) {
+    // MAKEFLAGS is not defined.
+    return result;
+  }
+
+  std::string err;
+  Jobserver::Config jobserver_config;
+  if (!Jobserver::ParseNativeMakeFlagsValue(makeflags, &jobserver_config,
+                                            &err)) {
+    // MAKEFLAGS is defined but could not be parsed correctly.
+    if (config_.verbosity > BuildConfig::QUIET)
+      status->Warning("Ignoring jobserver: %s [%s]", err.c_str(), makeflags);
+    return result;
+  }
+
+  if (!jobserver_config.HasMode()) {
+    // MAKEFLAGS is defined, but does not describe a jobserver mode.
+    return result;
+  }
+
+  if (config_.verbosity > BuildConfig::NO_STATUS_UPDATE) {
+    status->Info("Jobserver mode detected: %s", makeflags);
+#ifndef _WIN32
```

Comment:
This whole block can now be removed, correct?

Reply:
Done.